spec/requests/api/providers_spec.rb

```diff
@@ -130,6 +146,24 @@
       expect(response).to have_http_status(:forbidden)
     end
 
+    it "creates provider with custom attributes" do
+      api_basic_authorize collection_action_identifier(:providers, :create)
+      run_post(providers_url, sample_rhevm.merge!("custom_attributes" => custom_attributes))
```

Comment:
Not sure adding it as a sub-collection will work for me for few reasons:
1) I need a mass assignment option (getting lots of custom attributes in 1 request), currently i don't see its auto-supported.
2) I'm doing some overrides on the user input things like adding a section and check that the field-type is correct, I assume i can override the generic implementation just not sure if it worthwhile.
2) I need to give an option to send me the name of the custom attribute and update it on a provider based on name not by id.

@imtayadeway What you think ?